File: xmlIWebServisi/src/app/model/a1/CopyrightRequestDTO.ts

import { PodaciONaslovuDTO } from "./PodaciONaslovuDTO";
import { PodaciOPreradiDTO } from "./PodaciOPreradiDTO";
import { PodaciOAutorimaDTO } from "./PodaciOAutorimaDTO";
import { TLice } from "../TLice";

export class CopyrightRequestDTO{
    constructor(
        public podnosilacPrijave:TLice = new TLice(),
        public pseudonim:String = "",
        public punomocnik:TLice = new TLice(),
        public podaciONaslovu:PodaciONaslovuDTO = new PodaciONaslovuDTO(),
        public podaciOPreradi:PodaciOPreradiDTO = new PodaciOPreradiDTO(),
        public vrstaDela:String = "",
        public podaciOAutorima:PodaciOAutorimaDTO = new PodaciOAutorimaDTO(),
        public radnomOdnosu:boolean = false,
        public namenaDela:String = "",
        public potpis:String = "",
    ){}
}

File: BACKEND/autorska/src/main/java/com/example/autorska/dto/CopyrightRequestDTO.java
package com.example.autorska.dto;


import com.example.autorska.model.Autorska;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.xml.bind.annotation.XmlRootElement;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@XmlRootElement(name="autorskaPrijava")
public class CopyrightRequestDTO {

    private TLiceDTO podnosilacPrijave;

    private String pseudonim;

    private TLiceDTO punomocnik;

    private PodaciONaslovuDTO podaciONaslovu;

    private PodaciOPreradiDTO podaciOPreradi;

    private String vrstaDela;

    private PodaciOAutorimaDTO podaciOAutorima;

    private boolean radnomOdnosu;

    private String namenaDela;

    private String potpis;


    public CopyrightRequestDTO(Autorska autorska){
        this.podnosilacPrijave = new TLiceDTO(autorska.getPodnosilacPrijave());
        this.pseudonim = autorska.getPseudonim();
        this.punomocnik = new TLiceDTO(autorska.getPunomocnik());
        this.podaciONaslovu = new PodaciONaslovuDTO(autorska.getPodaciONaslovu());
        this.podaciOPreradi = new PodaciOPreradiDTO(autorska.getPodaciOPreradi());
    }



}
